### PR TITLE
First Commit in Matrix Library

### PR DESCRIPTION
**Pull request by AhmedMAtallah**

### CODE REVIEW

File: src/lib/matrixLibrary/Mat.h [hunk 1]
```diff
@@ -0,0 +1,150 @@
+#ifndef _MAT_H_
+#define _MAT_H_
+
+// all the system #include's we'll ever need
```
Comment (ammhassan) [inline src/lib/matrixLibrary/Mat.h]: This comment here doesn't seem to provide new information. Please remove it.

File: src/lib/matrixLibrary/Mat.h [hunk 2]
```diff
@@ -0,0 +1,150 @@
+#ifndef _MAT_H_
+#define _MAT_H_
+
+// all the system #include's we'll ever need
+#include <iostream>
+
```
Comment (ammhassan) [inline src/lib/matrixLibrary/Mat.h]: One of the Google C++ Style Guide rules is to minimize the use of vertical white spaces https://google.github.io/styleguide/cppguide.html#Vertical_Whitespace
Please format accordingly across all files.

File: src/lib/matrixLibrary/Mat.cpp [hunk 3]
```diff
@@ -0,0 +1,798 @@
...
+	Mat ainv(n,n);
+	for (i=0;i<n;i++) {
+		for (j=0;j<n;j++) ainv(i,j) = 0.;
+		ainv(i,i) = 1.;
+	}
+
+	int m=n_cols_;
+	
+	Mat xx(n),yy(n);
+	for (j=0;j<m;j++) {
```
Comment (ammhassan) [inline src/lib/matrixLibrary/Mat.cpp]: Please add space after the semicolon for all for loops.

File: src/lib/matrixLibrary/Mat.cpp [hunk 4]
```diff
@@ -0,0 +1,798 @@
...
+	//	cout << endl << " this " <<endl<<(*this)<<endl;
```
Comment (ammhassan) [inline src/lib/matrixLibrary/Mat.cpp]: Please remove the commented lines.

**Review by ammhassan — CHANGES_REQUESTED** [review]
Please remove `mat.o`

**Review by aatallah-ucsd — APPROVED** [review]

**Review by ammhassan — APPROVED** [review]

**Pull request merged**